### PR TITLE
[PROTOCOL 371] Improve the Natspec in TellerCommon

### PR DESCRIPTION
**Pull request by sypeer**
It is:
- Updated natspec for Teller Common

### CODE REVIEW

File: contracts/util/TellerCommon.sol [hunk 1]
```diff
@@ -10,23 +10,41 @@ import "./NumbersList.sol";
 library TellerCommon {
     enum LoanStatus {NonExistent, TermsSet, Active, Closed}
 
-    // The amount of interest owed to a borrower
-    // The interest is just that accrued until `timeLastAccrued`
+    /**
+        @notice The amount of interest owed to a borrower
+        @param totalAccruedInterest Total amount of accrued interest for a lender
+        @param totalNotWithdrawn The total amount of accrued interest claimed, but not yet withdrwan by the lender
```
Comment (hensha256) [inline contracts/util/TellerCommon.sol:16]: `withdrwan` -> `withdrawn`

File: contracts/util/TellerCommon.sol [hunk 2]
```diff
@@ -10,23 +10,41 @@ import "./NumbersList.sol";
...
     }
 
-    // Consensus request object for accruing interest
+    /**
+        @notice Consensus request object for accruing interest
+        @param lender The wallet address of the associated lender
+        @param consensusAddress The address of the Teller interest consensus contract to which the request is being submitted
+        @param requestNonce Nonce of the lender address used for authentication
+        @param startTime The timestamp at which the interest accrual is being requested to start
+        @param endTime The timestamp at which the interest accrual caluculation stops
```
Comment (hensha256) [inline contracts/util/TellerCommon.sol:45]: `caluculation` -> `calculation`

File: contracts/util/TellerCommon.sol [hunk 3]
```diff
@@ -36,7 +54,14 @@ library TellerCommon {
         uint256 requestTime;
     }
 
-    // Consensus response object for accruing interest
+    /**
+        @notice Consensus response object for accruing interest
+        @param signer The waller address of the signer validating the interest request of the lender
```
Comment (hensha256) [inline contracts/util/TellerCommon.sol:59]: `waller` -> `wallet`

File: contracts/util/TellerCommon.sol [hunk 4]
```diff
@@ -36,7 +54,14 @@ library TellerCommon {
...
+        @param consensusAddress The wallet address of the signer submitted the interest response of the submitted request
+        @param responseTime The timestamp at which the response was sent
+        @param interest The signed amount of accured interest earned that is caluculated by the signer
```
Comment (hensha256) [inline contracts/util/TellerCommon.sol:62]: `accured` -> `accrued`
`caluculated` -> `calculated`

**Review by Juan-cc — APPROVED** [review]

File: contracts/util/TellerCommon.sol [hunk 5]
```diff
@@ -36,7 +54,14 @@ library TellerCommon {
         uint256 requestTime;
     }
 
-    // Consensus response object for accruing interest
+    /**
+        @notice Consensus response object for accruing interest
+        @param signer The waller address of the signer validating the interest request of the lender
+        @param consensusAddress The wallet address of the signer submitted the interest response of the submitted request
```
Comment (hensha256) [inline contracts/util/TellerCommon.sol:60]: Missing a word between `signer` and `submitted` I think

File: contracts/util/TellerCommon.sol [hunk 6]
```diff
@@ -56,7 +90,16 @@ library TellerCommon {
         uint256 requestTime;
     }
 
-    // Borrower response object to take out a loan
+    /**
+        @notice Borrower response object to take out a loan
+        @param signer The waller address of the signer validating the interest request of the lender
```
Comment (hensha256) [inline contracts/util/TellerCommon.sol:95]: `waller` -> `wallet`

File: contracts/util/TellerCommon.sol [hunk 7]
```diff
@@ -56,7 +90,16 @@ library TellerCommon {
...
+        @param consensusAddress The address of the Teller loan consensus contract to which the request should be sent
+        @param responseTime The timestamp at which the response was sent
+        @param interestRate The signed interest rate generated by the signer's CRA
```
Comment (hensha256) [inline contracts/util/TellerCommon.sol:98]: Whats a CRA?

File: contracts/util/TellerCommon.sol [hunk 8]
```diff
@@ -67,14 +110,27 @@ library TellerCommon {
         Signature signature;
     }
 
-    // Represents loan terms based on consensus values
+    /**
+        @notice Represents loan terms based on consensus values
+        @param interestRate The consensus value for the interest rate based on all the loan responses from the signers
+        @param collateralRatio The consensus value for the he ratio of collateral to loan amount required for the loan, based on all the loan responses from the signers
```
Comment (hensha256) [inline contracts/util/TellerCommon.sol:116]: remove `he`

File: contracts/util/TellerCommon.sol [hunk 9]
```diff
@@ -84,7 +140,21 @@ library TellerCommon {
         uint256 duration;
     }
 
-    // Data per borrow as struct
+    /**
+        @notice Data per borrow as struct
+        @param id The id of the loan for internal tracking
+        @param loanTerms The loan terms returned by the signers
+        @param termsExpiry The timstamp at which the loan terms expire, after which if the loan is not yet active, cannot be taken out
```
Comment (hensha256) [inline contracts/util/TellerCommon.sol:147]: `timstamp` -> `timestamp`

File: contracts/util/TellerCommon.sol [hunk 10]
```diff
@@ -149,6 +223,11 @@ library TellerCommon {
     /**
         @notice This struct defines a market in the platform.
         @dev It is used by the MarketFactory contract.
+        @param loans The address for the Teller lLans contract that is being used for a market
```
Comment (hensha256) [inline contracts/util/TellerCommon.sol:226]: `lLans` -> `loans`

File: contracts/util/TellerCommon.sol (hunk 1 above)
Comment (sypeer) [reply contracts/util/TellerCommon.sol:16]: Fixed

File: contracts/util/TellerCommon.sol (hunk 2 above)
Comment (sypeer) [reply contracts/util/TellerCommon.sol:45]: Fixed

File: contracts/util/TellerCommon.sol (hunk 3 above)
Comment (sypeer) [reply contracts/util/TellerCommon.sol:59]: Fixed

File: contracts/util/TellerCommon.sol (hunk 4 above)
Comment (sypeer) [reply contracts/util/TellerCommon.sol:62]: Fixed

File: contracts/util/TellerCommon.sol (hunk 5 above)
Comment (sypeer) [reply contracts/util/TellerCommon.sol:60]: Fixed

File: contracts/util/TellerCommon.sol (hunk 6 above)
Comment (sypeer) [reply contracts/util/TellerCommon.sol:95]: Fixed

File: contracts/util/TellerCommon.sol (hunk 7 above)
Comment (sypeer) [reply contracts/util/TellerCommon.sol:98]: Credit Risk Algorithm (CRA) - made the change

File: contracts/util/TellerCommon.sol (hunk 8 above)
Comment (sypeer) [reply contracts/util/TellerCommon.sol:116]: Removed

File: contracts/util/TellerCommon.sol (hunk 9 above)
Comment (sypeer) [reply contracts/util/TellerCommon.sol:147]: Fixed

File: contracts/util/TellerCommon.sol (hunk 10 above)
Comment (sypeer) [reply contracts/util/TellerCommon.sol:226]: Changed

**Review by hensha256 — APPROVED** [review]

**Review by passabilities — APPROVED** [review]

**Review by salazarguille — APPROVED** [review]

**Pull request merged**